### DISCUSSION TITLE
Optional passing for null?

### DISCUSSION THREAD

**Discussion by m1212e** [sinclairzx81/typebox#759] [Q&A] [answered]
Hi, I use a generator to build typebox schemas from my prisma schema. Optional fields from Prisma are returned as value or null. But the typebox typescript type which is derived from the use with `Static<>` results in a value or undefined. So when I make a DB call I get a type of e.g. `string | null` for an optional field but the typescript compiler complains because the type derived from the schema which uses `Optional()` results in `string | undefined` which is a type mismatch. Can I somehow easily resolve this without having to re map the returned types all the time?

Thanks!

**Comment by sinclairzx81** [comment, marked as answer]
@m1212e Hi,

> So when I make a DB call I get a type of e.g. string | null for an optional field but the typescript compiler complains because the type derived from the schema which uses Optional() results in string | undefined which is a type mismatch. Can I somehow easily resolve this without having to re map the returned types all the time?

Try the following

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxjsriQBYYgFwC+cAZlBCHAOQACAzsADsSAGwLAoAehjoMAIwgAPTgChlJCAN7wAcgFdho2cIxwAvHAA8yOBgUwMAgCa8UlGnQB8ACl7VaBAC4UAEozDxQZADoAVQFgDS8AbTRMSL0DL2CcX3cCAF1g1XVNeGtzFIxIgHlZACsMEhgvBGU4NoECEAwg9MNjLwrI-ChBAHNMrNa24wA3DGEe-T6MAai9EFkMKAmsKZsOowxHRYMCQ9XUgCEICGMCAQnlBkLlaUwUMzxCYhILN4wIMwUB4gA)

```typescript
import { Type, Static, TSchema } from '@sinclair/typebox'

const Nullable = <T extends TSchema>(schema: T) => Type.Union([Type.Null(), schema])

const T = Type.Object({
   name: Nullable(Type.String()),
   level: Nullable(Type.Number()),
   enabled: Nullable(Type.Boolean())
})

type T = Static<typeof T>
```
Just keep in mind that Optional means "the property key is optional" (so is somewhat unrelated to `null` or `undefined`). The above type has the expectation that all keys must exist, and the property value for each key must be either `T | null` (as denoted by the union)

Hope this helps
S

**Reply by m1212e** [reply]
Ah I see. Thank you very much!

**Reply by woss** [reply]
i think this should be in the readme if it's not already

**Reply by louwers** [reply]
This should be part of the library I think.

**Reply by louwers** [reply]
Info here for those wondering: https://github.com/sinclairzx81/typebox/issues/989

**Reply by rp-pedraza** [reply]
Is there a way to do this to all properties much like Partia?
